nibabel/fileslice.py

```diff
@@ -634,29 +634,49 @@ def read_segments(fileobj, segments, n_bytes):
         absolute file offset in bytes and number of bytes to read
     n_bytes : int
         total number of bytes that will be read
+    lock : threading.Lock
+        If provided, used to ensure that paired calls to ``seek`` and ``read``
+        cannot be interrupted by another thread accessing the same ``fileobj``.
+        Each thread which accesses the same file via ``read_segments`` must
+        share a lock in order to ensure that the file access is thread-safe.
+        A lock does not need to be provided for single-threaded access.
 
     Returns
     -------
     buffer : buffer object
         object implementing buffer protocol, such as byte string or ndarray or
         mmap or ctypes ``c_char_array``
     """
+    # Make a dummy lock-like thing to make the code below a bit nicer
+    if lock is None:
+
+        class DummyLock(object):
```

Comment:
Push this out of the method to the module top-level, maybe as `class _NullLock` ?